### ISSUE TITLE
Add support for non-JSON image mocks

### ISSUE THREAD

**Issue by etpinard** [plotly/plotly.js#913]
PR https://github.com/plotly/plotly.js/pull/850 must us realised that not all plotly.js features are JSON serializable e.g. `Float32Array`.

**Possible solution:** we could try to make the image tests routine and the test dashboard accept `js` files as mock. These `js` files would generate `{ data: [], layout: {} }` object.

For example,

```js
// in test/image/mocks/float32array.js

module.exports = function() {
  return {
    data: [{
      xy: new Float32Array([1,2,3,4])
    }]
  }
}
```

**Comment by etpinard** [comment]
Ping @plotly/plotly_js 

Might be a good time to tackle this.

**Comment by etpinard** [comment]
So, I propose:

- keep the `test/image/mocks/` folder intact. A lot of things (e.g. https://rreusser.github.io/plotly-mock-viewer/) are looking into this folder from `http://raw.githubusercontent.com/` to do stuff with the JSON mocks.
- add a `test/image/mock-generators/` folder with js files such that: `test/image/mock-generators/some-mock.js` would correspond to `test/image/mocks/some-mock.json`

then

- before `npm run test-image`, `npm run baseline` and `npm run test-export` we would run these `test/image/mock-generators/` scripts in Node.js to then update `text/image/mocks/` before generating baselines or running the image tests.

Now, we would still need to keep the output of `test/image/mock-generators/` pure JSON (e.g. no TypedArrays). Down the road (when / if we switch to `orca` to run our image tests), we could try to generate the mocks in the front-end by adding a new `orca` renderer.

We would also need to find a way to keep `test/image/mock-generators/` and `test/image/mocks/` in-sync. I can think of two solutions:

+ PRs that add files to `test/image/mock-generators/` **do not add** the corresponding files in `test/image/mocks/`. `test/image/mocks/` would be added during version commits.
  - :+1: no extra step (that could be sometimes forgotten) for developers
  - :-1: new `test/image/mocks/` JSON files are not available until next plotly.js version

+ PRs that add files to `test/image/mock-generators/`, **must add** the corresponding files in `test/image/mocks/`
  - :+1:  `test/image/mocks/` JSON files are always available
  - :-1:  we would have to add extra machinery (probably a syntax test) to make sure `test/image/mock-generators/` and `test/image/mocks/` are in-sync

Thougths?

**Comment by archmoj** [comment]
I'd vote for the second solution noting that few jasmine tests are dependent on the mocks; and one may want to run them in parallel.

**Comment by gvwilson** [comment]
Hi - this issue has been sitting for a while, so as part of our effort to tidy up our public repositories I'm going to close it. If it's still a concern, we'd be grateful if you could open a new issue (with a short reproducible example if appropriate) so that we can add it to our stack. Cheers - @gvwilson
